STM32: Replace HAL_GetTick

bcostm opened the pull request:
### Description

- Replace the default ST HAL_GetTick function (defined as weak) in order to use the ST HAL_Delay even when interrupts are disabled. Before a variable was incremented at each us_ticker interrupt and read back by the HAL_GetTick function. Now this is no more needed, we read directly the us_ticker value.
- Remove also all the code related to timer channel 2 (no more used)

Tested with a basic code using together HAL_Delay() and _disable_irq functions. 

This method has been proposed as a patch by @c1728p9  in PR #6555 

### Pull request type

    [X] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

bcostm committed:
Replace HAL_GetTick

LMESTM commented on targets/TARGET_STM/hal_tick_32b.c:
```diff
@@ -44,8 +42,6 @@ void timer_irq_handler(void)
             __HAL_TIM_CLEAR_IT(&TimMasterHandle, TIM_IT_CC2);
```
I like the idea - great PR ! 
now if the HAL_GetTick relies on common ticker layer, then I think that the complete code related to the usage of TIM_IT_CC2 can be removed I guess - don't you think ? 
This would mean no need to enable/disable CC2 IT, to need to manage the interrupt anymore....

bcostm replied:
Yes I think you're right. The timer channel2 was here only to generate an interrupt for the HAL tick. I'll make a try.

bcostm replied:
It works: I have removed all the code related to timer channel 2 and all mbedOS5 timer tests are PASS. Tested with both 16-bits and 32-bits timers.

LMESTM reviewed (commented):
I think we can remove much more code with this proposal - what do you think ?

bcostm committed:
Remove code related to timer channel 2

bcostm committed:
Cleanup

- Remove calls to HAL_SuspendTick and HAL_ResumeTick
- Rename stm_common.c in hal_tick_common.c

bcostm committed:
Remove HAL_TICK_DELAY (no more used)

0xc0170 commented on targets/TARGET_STM/hal_tick_common.c:
```diff
@@ -0,0 +1,18 @@
+#include "hal/us_ticker_api.h"
```
Can you add a license header on top of this file?

bcostm replied:
Yep. Forgot it...

bcostm committed:
Add license

0xc0170 commented:
/morph build

cmonr commented:
Halting CI builds until RC3 PRs are completed. Will resume after.

0xc0170 commented:
/morph test
/morph export-build
